Odata-docs/client/bulk-update-operations.md

```diff
@@ -0,0 +1,245 @@
+
+``` csharp
+public class Customer
+{
+    public int Id { get; set; }
+    public string Name { get; set; }
+    public int Age { get; set; }
+    public List<Order> Orders { get; set; }
+}
+
+public class Order
+{
+    public int Id { get; set; }
+    public decimal Amount { get; set; }
+}
+```
+### Sample Request 1
+This sample request demonstrates how to create a bulk update request in OData client to create 2 new customers in a single request. 
+
+```http
+PATCH http://localhost:6285/odata/Customers
+```
+Request Body
+
+```json
+{
+	"@context": "http://localhost:6285/odata/$metadata#Customers/$delta",
+	"value": [
+		{
+			"Id": 1,
+			"Name": "Customer1",
+			"Age": 26
+		},
+		{
+			"Id": 2,
+			"Name": "Customer2",
+			"Age": 34
+		}
+	]
+}
+```
+
+How to make the above request from OData client using OData Connected Service. 
+
+```csharp
+var context = new Container(new Uri("http://localhost:6285/odata/"));
+
+var customer1 = new Customer
+{
+    Id = 1,
+    Name = "Customer1",
+    Age = 26
+};
+
+context.AddToCustomers(customer1);
+
+var customer2 = new Customer
+{
+    Id = 2,
+    Name = "Customer2",
+    Age = 34
+};
+
+context.AddToCustomers(customer2)
+
+DataServiceResponse response = await context.BulkUpdateAsync(customer1, customer2);
+```
+
+### Sample Request 2
+This sample request demonstrates how to create a bulk update request to create a new customer and 2 new orders and link the 2 created orders to the created customer.  
+
+```http
+PATCH http://localhost:6285/odata/Customers
+```
+Request Body
+
+```json
+{
+	"@context": "http://localhost:6285/odata/Customers/$delta",
+	"value": [{
+		"Id": 1,
+		"Name": "Customer1",
+		"Orders@delta": [{
+				"Id": 1,
+				"Amount": 10.24,
+			},
+			{
+				"Id": 2,
+				"Amount": 11.54
+			}
+		]
+	}]
+}
+```
+
+How to make the above request from OData client using OData Connected Service.
+
+```csharp
+var context = new Container(new Uri("http://localhost:6285/odata/"));
+
+var customer = new Customer
+{
+    Id = 1,
+    Name = "Customer1",
+    Age = 26
+};
+
+context.AddToCustomers(customer1);
+
+var order = new Order
+{
+    Id = 2,
+    Amount = 10.24
+};
+
+var order1 = new Order
+{
+    Id = 3,
+    Amount = 11.54
+}
+
+context.AddToCustomers(customer2);
+
+context.AddRelatedOject(customer, "Orders", order);
+context.AddRelatedObject(customer, "Orders", order1);
+
+DataServiceResponse response = await context.BulkUpdateAsync(customer);
+```
+
+### Sample Request 3
+This sample request demonstrates how to create a new customer and update the amounts of 2 existing orders then link the updated orders to the created customer.
+
+```http
+PATCH http://localhost:6285/odata/Customers
+```
+Request Body
+```json
+{
+	"@context": "http://localhost:6285/odata/Customers/$delta",
+	"value": [{
+		"Id": 1,
+		"Name": "Customer1",
+		"Orders@delta": [
+            {
+				"@id":"Orders(1)", 
+                "Amount": 14.12
+			},
+			{
+				"@Id": "Orders(2)",
```

Comment:
Does it matter that this is `@Id` while for the first order it's `@id`? It probably doesn't but it's better to consistent so that readers don't face similar questions...